### PR TITLE
Hotfix/eigcg fix

### PR DESCRIPTION
**Pull request by alexstrel**
This is a bug-fix pull request for the eigcg solver: 
-- added check when the number of low modes exceeds the entire search space, i.e., when `2*param.nev > param.m`. 
-- updated `deflated_invert_test.cpp` .

### CODE REVIEW

File: lib/inv_eigcg_quda.cpp [hunk 1]
```diff
@@ -252,6 +252,9 @@ namespace quda {
   IncEigCG::IncEigCG(DiracMatrix &mat, DiracMatrix &matSloppy, DiracMatrix &matPrecon, SolverParam &param, TimeProfile &profile) :
     Solver(param, profile), mat(mat), matSloppy(matSloppy), matPrecon(matPrecon), K(nullptr), Kparam(param), Vm(nullptr), r_pre(nullptr), p_pre(nullptr), eigcg_args(nullptr), profile(profile), init(false)
   {
+
+    if( 2*param.nev > param.m )  errorQuda("\nIncorrect number of the requested low modes: m= %d while nev=%d (note that 2*nev must be less then m).\n", param.m, param.nev);
```
Comment (maddyscientist) [inline lib/inv_eigcg_quda.cpp]: No need for trailing `\n` since `errorQuda` includes this

**Comment by alexstrel** [comment]
Hi @maddyscientist , thanks for comments, done.

**Comment by maddyscientist** [comment]
This looks good @alexstrel.  @hummingtree have you tested with these changes to see if it addresses the issues you had?

**Comment by maddyscientist** [comment]
Merging this pull.  While Mobius with eigCG is not working robustly, this pull does correct the urgent issues observed.

**Pull request merged**